plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -620,13 +479,167 @@ function ( $value ) {
 		);
 	}
 
+	/**
+	 * Returns the keys of all core fields.
+	 *
+	 * @return array An array of field keys.
+	 */
+	public function get_core_fields_keys() {
+		return [
+			'email',
+			'first_name',
+			'last_name',
+			'company',
+			'address_1',
+			'address_2',
+			'country',
+			'city',
+			'state',
+			'postcode',
+			'phone',
+		];
+	}
+
```

Comment:
I see why this is added, so that `fields_locations` can be populated in the constructor, but it's obviously not ideal since it means the keys have to be updated in two places if they change. But, since both the `core_fields` and the `fields_locations` properties in this class are set to private, we could do the same treatment for `fields_locations`, i.e. create a new private method called `get_fields_locations` and use that instead of referencing the class property. Then you could remove the property assignment from the class constructor.

Reply:
Agreed, the duplication is not ideal. A slight difference here is that `fields_locations` can be mutated via `register_checkout_field()`/`deregister_checkout_field`. Plus, it's accessed in quite a few places throughout the class. Any suggestions on how best to handle that?